include/mgp.hpp

```diff
@@ -3547,4 +3563,60 @@ template <>
 struct hash<mgp::MapItem> {
   size_t operator()(const mgp::MapItem &x) const { return hash<std::string_view>()(x.key); };
 };
+
+template <>
+struct hash<mgp::Value> {
+  size_t operator()(const mgp::Value &value) const {
+    switch (value.Type()) {
+      case mgp::Type::Null:
+        return 31;
+      case mgp::Type::Any:
+        throw mg_exception::InvalidArgumentException();
+      case mgp::Type::Bool:
+        return std::hash<bool>{}(value.ValueBool());
+      case mgp::Type::Int:
+        // we cast int to double for hashing purposes
+        // to be consistent with equality (2.0 == 2) == true
+        return std::hash<double>{}((double)value.ValueInt());
+      case mgp::Type::Double:
+        return std::hash<double>{}(value.ValueDouble());
+      case mgp::Type::String:
+        return std::hash<std::string_view>{}(value.ValueString());
+      case mgp::Type::List:
+        return memgraph::utils::FnvCollection<mgp::List, mgp::Value, std::hash<mgp::Value>>{}(value.ValueList());
+      case mgp::Type::Map:
+        return memgraph::utils::FnvCollection<mgp::Map, mgp::MapItem, std::hash<mgp::MapItem>>{}(value.ValueMap());
+      case mgp::Type::Node:
+        return std::hash<mgp::Node>{}(value.ValueNode());
+      case mgp::Type::Relationship:
+        return std::hash<mgp::Relationship>{}(value.ValueRelationship());
+      case mgp::Type::Path: {
+        // https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function
+        // See memgraph::utils::FnvCollections above for more info
+        constexpr const uint64_t fnv_prime = 1099511628211U;
+        uint64_t hash = 14695981039346656037U;
+
+        const auto &path = value.ValuePath();
+        for (size_t i = 0; i < path.Length() - 1; ++i) {
+          hash *= fnv_prime;
+          hash ^= std::hash<mgp::Node>{}(path.GetNodeAt(i));
+          hash *= fnv_prime;
+          hash ^= std::hash<mgp::Relationship>{}(path.GetRelationshipAt(i));
+        }
+        hash *= fnv_prime;
+        hash ^= std::hash<mgp::Node>{}(path.GetNodeAt(path.Length()));
+        return hash;
```

Comment:
Can you isolate this in function of some sort, or at least this part in for loop, so you have function hashPathValue which returns hash